diff --git a/bridge/matrix.go b/bridge/matrix.go
--- a/bridge/matrix.go
+++ b/bridge/matrix.go
@@ -607,7 +607,16 @@
 	}
 
 	user := mx.bridge.Child.GetIUser(evt.Sender, true)
-	if user == nil || user.GetPermissionLevel() < bridgeconfig.PermissionLevelUser || !user.IsLoggedIn() {
+	if user == nil || !user.IsLoggedIn() {
+		if child, ok := mx.bridge.Child.(interface {
+			GetLoggedInUserForPortal(id.RoomID) User
+		}); ok {
+			user = child.GetLoggedInUserForPortal(evt.RoomID)
+		}
+		if user == nil {
+			return
+		}
+	} else if user.GetPermissionLevel() < bridgeconfig.PermissionLevelUser {
 		return
 	}
 
