diff --git a/pkg/msgconv/matrixfmt/blocks.go b/pkg/msgconv/matrixfmt/blocks.go
--- a/pkg/msgconv/matrixfmt/blocks.go
+++ b/pkg/msgconv/matrixfmt/blocks.go
@@ -401,6 +401,9 @@ func (parser *HTMLParser) textToElements(text string, ctx Context) []slack.Rich
 	if ctx.TagStack.Has("code") || ctx.TagStack.Has("pre") || ctx.TagStack.Has("a") {
 		return []slack.RichTextSectionElement{parser.textToElement(text, ctx)}
 	}
+	if ctx.Mentions != nil && ctx.Mentions.Room && !strings.Contains(text, "@room") {
+		text = "@room " + text
+	}
 	var pattern *regexp.Regexp
 	if ctx.Mentions != nil && ctx.Mentions.Room {
 		pattern = URLOrRoomRegex
@@ -419,7 +422,7 @@ func (parser *HTMLParser) textToElements(text string, ctx Context) []slack.Rich
 			elems = append(elems, parser.textToElement(prefix, ctx))
 		}
 		if part == "@room" {
-			elems = append(elems, slack.NewRichTextSectionBroadcastElement(slack.RichTextBroadcastRangeChannel))
+			elems = append(elems, slack.NewRichTextSectionBroadcastElement(slack.RichTextBroadcastRangeHere))
 		} else if strings.HasPrefix(part, "http://") || strings.HasPrefix(part, "https://") {
 			elems = append(elems, slack.NewRichTextSectionLinkElement(part, part, ctx.StylePtr()))
 		} else {
diff --git a/pkg/msgconv/mrkdwn/tag.go b/pkg/msgconv/mrkdwn/tag.go
--- a/pkg/msgconv/mrkdwn/tag.go
+++ b/pkg/msgconv/mrkdwn/tag.go
@@ -253,7 +253,7 @@ func (r *slackTagHTMLRenderer) renderSlackTag(w goldmarkUtil.BufWriter, source [
 			}
 			return
 		case "channel", "everyone", "here":
-			// do @room mentions?
+			_, _ = w.WriteString("@room")
 			return
 		case "subteam":
 			// do subteam handling? more spaces?
